src/Components/Admin/Admin.js

```diff
@@ -30,14 +53,33 @@ class Admin extends React.Component {
    * @param {*} password is the password of the person logging in
    */
   authenticate(username, password) {
-    //write some legit authentication logic in here eventually
-    //if authentication passes then call readFromDatabase(); otherwise not needed
-    this.setState({
-      loggedIn: true,
-      username: username,
-      password: password
-    });
-  }
+    if(username.charAt('@') === -1 || username.charAt('.') === -1)
+      return;
+  db.collection("Student").where('Email', '==', username).get()
+  .then(snapshot => {
+    if (snapshot.empty) {
+      console.log('No matching documents.');
+      return;
+    }
+    else{
+      snapshot.forEach(doc => {
+        var data = doc.data();
+        if(data.Password === password)
+        {
+          this.setState({
+            loggedIn: true,
+            username: data.Email,
+            password: data.Password,
+            id: data.id,
+            isAdmin: data.isAdmin
+          })
+        }
+      });
+    }
+     }).catch(function(error) {
+       console.log("Error getting document:", error);
+      })
```

Comment:
At this point we can also add in check to see if they got successfully logged in else display invalid error message in UI
if(!this.state.loggedIn) { //display invalid username/password in UI; }